paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -33,6 +37,199 @@ using framework::Tensor;
+    auto src_md = dnnl::memory::desc(
+        src_dims, framework::ToMKLDNNDataType(tensor_src->type()),
+        framework::ToMKLDNNFormat(current_layout));
+    auto dst_md = dnnl::memory::desc(
+        src_dims, framework::ToMKLDNNDataType(tensor_src->type()),
+        framework::ToMKLDNNFormat(target_layout));
+    auto src_mem = dnnl::memory(
+        src_md, engine,
+        reinterpret_cast<void*>(const_cast<T*>(tensor_src->data<T>())));
+    auto dst_mem = dnnl::memory(dst_md, engine, out);
+    // Create primitive descriptor.
+    auto reorder_pd =
+        dnnl::reorder::primitive_desc(engine, src_md, engine, dst_md);
+    // Create the primitive.
+    auto reorder_prim = dnnl::reorder(reorder_pd);
+    // Primitive arguments.
+    std::unordered_map<int, dnnl::memory> reorder_args;
+    reorder_args.insert({DNNL_ARG_SRC, src_mem});
+    reorder_args.insert({DNNL_ARG_DST, dst_mem});
+    // Primitive execution: reorder with scaled sum.
+    reorder_prim.execute(engine_stream, reorder_args);
+    // Wait for the computation to finalize.
+    engine_stream.wait();
+  }
+
+ public:
+  static void execute(const std::string& label, const std::string& name,
+                      const framework::Tensor& _tensor) {
+    // TensorDumpConfig();
+    auto it = TensorDumpConfig::get().fetchOperator(label);
+    if (it == TensorDumpConfig::get().fetchOperatorEnd()) {
+      return;
+    }
+    auto target_layout = TensorDumpConfig::get().getOperatorLayout(label);
+    std::string filename = TensorDumpConfig::get().getFoldername() + label +
+                           "_" + name + "_" +
+                           framework::DataTypeToString(_tensor.type());
+
+    if (target_layout != OperatorDetails::getDefaultLayout()) {
+      filename = filename + "_" + DataLayoutToString(target_layout);
+
+      std::vector<T> vout(_tensor.numel());
+      reorder_via_mkldnn(vout.data(), &_tensor, target_layout);
+
+      auto data_limit = std::numeric_limits<int64_t>::max();
+      if (const char* value = std::getenv("DUMP_LIMIT")) {
+        data_limit = atoi(const_cast<char*>(value));
+      }
+      data_limit = data_limit < _tensor.numel() ? data_limit : _tensor.numel();
+
+      std::lock_guard<decltype(TensorDumpConfig::getMutex())> l(
+          TensorDumpConfig::getMutex());
+      auto ofs = std::ofstream(filename, std::ios_base::app);
+      ofs << SeqInFile::Access(filename) << std::endl;
+      print_memory(ofs, vout, data_limit);
+    } else {
+      filename = filename + "_" + DataLayoutToString(target_layout);
+
+      /* in case of parallel executor , io must be synchronized */
+      std::lock_guard<decltype(TensorDumpConfig::getMutex())> l(
+          TensorDumpConfig::getMutex());
+      std::ofstream(filename, std::ios_base::app) << SeqInFile::Access(filename)
+                                                  << std::endl
+                                                  << _tensor << std::endl;
+    }
+  }
+};
+
```

Comment:
I think this is not the best place for this code. Please move it to separate file and put in a specific namespace.